extendr-api/src/wrapper/altrep.rs

```diff
@@ -455,6 +455,16 @@ pub trait AltStringImpl {
     }
 }
 
+#[cfg(use_r_altlist)]
+pub trait AltListImpl {
+    /// Get a single element from this vector
+    // a single element of a list can be any Robj
+    fn elt(&self, _index: usize) -> Robj;
+
+    /// Set a single element in this list.
+    fn set_elt(&mut self, _index: usize, _value: Robj) {}
```

Comment:
Why do we have a default implementation for `set_elt()` here?

Reply:
I think this just follows other altrep-imlementation. 
I cannot decide which is good, but making it read-only by default is probably not a very strange idea.

Reply:
But it is not readonly? If you implement a trait with default method implementation, you get access to this method anyway? Or you can pass around this type as `T : Trait` or `impl Trait`, which exposes all methods. In which case, settings becomes a `noop`, which could be misleading. If it is our go-to approach, then no objection, but it is worth investigating how it works and why we chose this approach.

Reply:
I have no idea, sorry. My main point was that this isn't @JosiahParry's fault so we should not block for this reason.

> it is worth investigating how it works and why we chose this approach.

I agree with you. Let's merge this and discuss in another issue.

Reply:
This is copying evrey other altrep trait implementation. The `set_elt()` and `elt()` are required for the ALTLIST.

I think the atlrep story is unfnished and can be improved upon a lot. This is a baby step PR towards that direction.